ecommerce/HooksHandler.php

```diff
@@ -405,5 +409,86 @@ public function manage_earnings_and_enrollments( string $order_status, int $orde
 		$earnings->prepare_order_earnings( $order_id );
 		$earnings->remove_before_store_earnings();
 	}
+
+	/**
+	 * Update order data for the free checkout
+	 *
+	 * @since 3.4.0
+	 *
+	 * @param array $order_data Order data.
+	 *
+	 * @return array
+	 */
+	public function update_order_data( array $order_data ) {
+		if ( empty( $order_data['total_price'] ) ) {
+			$order_data['order_status']   = OrderModel::ORDER_COMPLETED;
+			$order_data['payment_status'] = OrderModel::PAYMENT_PAID;
+			$order_data['payment_method'] = 'free';
+		}
+		return $order_data;
+	}
+
+	/**
+	 * Enroll user to the course when free checkout
+	 *
+	 * @since 3.4.0
+	 *
+	 * @param array $order_data Order data.
+	 *
+	 * @return array
+	 */
+	public function handle_free_checkout( array $order_data ) {
+		if ( empty( $order_data['total_price'] ) ) {
```

Comment:
We need to check order type and if plan subscription need to check it is course or bundle subscription.